core/api/service/author/impl/author_api_impl.cpp

```diff
@@ -96,4 +104,30 @@ namespace kagome::api {
     return outcome::failure(boost::system::error_code{});
   }
 
+  outcome::result<AuthorApi::SubscriptionId>
+  AuthorApiImpl::submitAndWatchExtrinsic(Extrinsic extrinsic) {
+    extrinsic.observed_id = latest_id_++;
+    subscribed_ids_.insert(extrinsic.observed_id.value());
+    OUTCOME_TRY(submitExtrinsic(extrinsic));
+    if (auto service = api_service_.lock()) {
+      OUTCOME_TRY(sub_id,
+                  service->subscribeForExtrinsicLifecycle(
+                      extrinsic.observed_id.value()));
+      return sub_id;
+    }
+    throw jsonrpc::InternalErrorFault(
+        "Internal error. Api service not initialized.");
+  }
+
+  outcome::result<bool> AuthorApiImpl::unwatchExtrinsic(SubscriptionId sub_id) {
+    if (auto service = api_service_.lock()) {
+      OUTCOME_TRY(service->unsubscribeFromExtrinsicLifecycle(sub_id));
+    } else {
+      throw jsonrpc::InternalErrorFault(
+          "Internal error. Api service not initialized.");
+    }
+    bool existed = subscribed_ids_.erase(sub_id) > 0;
+    return existed;
```

Comment:
You can return ```result<bool>``` from ```unsubscribeFromExtrinsicLifecycle``` and from ```unsubscribe``` in ```Subscriber``` => You can remove ```subscribed_ids_```. And function become:
```c++
    if (auto service = api_service_.lock())
      return service->unsubscribeFromExtrinsicLifecycle(sub_id);

    throw jsonrpc::InternalErrorFault(
        "Internal error. Api service not initialized.");

Reply:
Fair enough.